Let compile on x86

IslandUsurper opened the pull request:
Fixes #3. Uses types from `std::arch::x86` as necessary. Also adds check for `sse2` feature.

~Completely untested as yet.~

IslandUsurper committed:
Let compile on x86

Uses types from `std::arch::x86` as necessary.

jrmuizel commented:
Instead of duplicating all the uses maybe we can conditionally alias x86/x86_64 as x86_intrins or something like that and only use `x86_intrins::{ _mm_loadu_si128, _mm_storeu_si128}` etc.

IslandUsurper commented:
Fixed a typo, and added another cfg(target_feature). Tests now pass with target i686-unknown-linux-musl.

Do you think an alias would be easier to read, @jrmuizel? I guess we can try it and see.

IslandUsurper committed:
Conditionally compile tests. Alias arch modules.

IslandUsurper commented:
You're right. I like that better too.

I'm a bit concerned that `over_in_sse2()` isn't covered by a test, but I don't know enough about compositing or SIMD to write one. And, that might be a separate issue anyway.

jrmuizel commented on src/lib.rs:
```diff
@@ -678,15 +680,21 @@ pub fn over_in(src: u32, dst: u32, alpha: u32) -> u32 {
+        _mm_loadu_si128,
+        _mm_storeu_si128,
+    };
+    #[cfg(target_arch = "x86_64")]
```
This piece should use the same x86_intrinsics stuff from below. Other than that this looks good.

IslandUsurper replied:
Good point.
